backend/bot/index.js

```diff
@@ -101,22 +96,43 @@ const postRankingChangeMessage = async (oldRankings, newRankings) => {
       `${c.name} ${c.oldRanking}. ⟶ ${c.newRanking}.`)
     )
     .join('\n')
-  await bot.postMessage(channel.id, messageText, { as_user: true });
+  return messageText
 }
 
 const hasLeaderboardChanged = (leaderboardSize, oldRankings, newRankings) => (
   oldRankings.findIndex((oldPlayer, index) => oldPlayer.id !== newRankings[index].id) < leaderboardSize
 )
 
-const updatePurpose = async (rankings) => {
+const createPurposeMessage = async (rankings) => {
   const rankingsText = rankings
     .map((ranking, i) => `${i + 1}. ${ranking.name} (${ranking.rating})`)
     .join('\n')
   const purpose = 'TOP PLAYERS\n' + rankingsText
-  await bot.setGroupPurpose(channel.id, purpose)
+  return purpose
 }
 
 
+const makeBot = async (foosbotToken, channelName) => {
+  return new Promise((resolve, reject) => {
+    if (!foosbotToken || !channelName) {
+      return reject("Missing env variables!")
+    }
+
+    const slackBot = new SlackBot({
+      token: foosbotToken
+    })
+    
+    slackBot.on('start', async () => {
+      console.log('Foosbot started')
```

Comment:
I would print this message only when the bot initializes correctly, preferrably into `match-repository.js` under the call of `makeBot`. In other words - when `makeBot` resolves.

Reply:
good idea, thanks
i put it in app.js